src/main/java/net/minecraftforge/client/model/ItemTextureQuadConverter.java

```diff
@@ -325,10 +327,22 @@ private static void putVertex(UnpackedBakedQuad.Builder builder, VertexFormat fo
                     if (format.getElement(e).getIndex() == 0)
                     {
                         builder.put(e, u, v, 0f, 1f);
-                        break;
                     }
+                    break;
                 case NORMAL:
-                    builder.put(e, (float) side.getFrontOffsetX(), (float) side.getFrontOffsetY(), (float) side.getFrontOffsetZ(), 0f);
+                    if (hasTransform)
+                    {
+                        vec.x = (float) side.getFrontOffsetX();
```

Comment:
Any way this logic could be deduped? It might be too much for this PR, but should probably be done at some point.

Reply:
I do agree about the duplication, while there are a few minor differences, we should try to unify it more.
Perhaps it can be looked at as part of the 1.13 update?

Reply:
For now, some of the duplication is fixed by 9e4b9e6d7a81963def44ef32b7f3ab95038e3ee4.